code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -113,6 +113,89 @@
 		return
 	return held_item.GetID()
 
+/mob/living/carbon/human/proc/get_accessible_cash()
+	var/available_cash = 0
+	var/list/cash_list = get_cash_list()
+	if(!length(cash_list))
+		return 0
+	for(var/found_item in cash_list)
+		if(istype(found_item, /obj/item/holochip))
+			var/obj/item/holochip/chip_stack = found_item
+			available_cash += chip_stack.credits
+		if(istype(found_item, /obj/item/card/id))
+			var/obj/item/card/id/id_card = found_item
+			available_cash += id_card.registered_account.account_balance
+	return available_cash
+
+/mob/living/carbon/human/proc/spend_cash(var/to_spend)
```

Comment:
I feel like this can fail and eat your money when your holochip isn't sufficient but somehow you make to purchase because this doesn't check if you can purchase first and spends your money first.
you shpuld check the whole cash list to to_spend at first, and re-do this once you're eligible to purchase.

Reply:
It already does that